A few compilation warnings

```
[233/531] Building CXX object solver/hydro/CMakeFiles/antares-solver-hydro.dir/management/finalLevelValidator.cpp.o
In file included from /home/florian/Antares_Simulator/src/solver/hydro/management/finalLevelValidator.cpp:28:
/home/florian/Antares_Simulator/src/solver/hydro/include/antares/solver/hydro/management/finalLevelValidator.h: In constructor ‘Antares::Solver::FinalLevelValidator::FinalLevelValidator(Antares::Data::PartHydro&, unsigned int, Antares::Data::AreaName, double, double, unsigned int, unsigned int, unsigned int, Antares::HydroErrorsCollector&)’:
/home/florian/Antares_Simulator/src/solver/hydro/include/antares/solver/hydro/management/finalLevelValidator.h:74:35: warning: ‘Antares::Solver::FinalLevelValidator::areaName_’ will be initialized after [-Wreorder]
   74 |     const Antares::Data::AreaName areaName_;
      |                                   ^~~~~~~~~
/home/florian/Antares_Simulator/src/solver/hydro/include/antares/solver/hydro/management/finalLevelValidator.h:73:18: warning:   ‘unsigned int Antares::Solver::FinalLevelValidator::areaIndex_’ [-Wreorder]
   73 |     unsigned int areaIndex_;
      |                  ^~~~~~~~~~
/home/florian/Antares_Simulator/src/solver/hydro/management/finalLevelValidator.cpp:33:1: warning:   when initialized here [-Wreorder]
   33 | FinalLevelValidator::FinalLevelValidator(
      | ^~~~~~~~~~~~~~~~~~~
/home/florian/Antares_Simulator/src/solver/hydro/include/antares/solver/hydro/management/finalLevelValidator.h:76:12: warning: ‘Antares::Solver::FinalLevelValidator::finalLevel_’ will be initialized after [-Wreorder]
   76 |     double finalLevel_;
      |            ^~~~~~~~~~~
/home/florian/Antares_Simulator/src/solver/hydro/include/antares/solver/hydro/management/finalLevelValidator.h:67:18: warning:   ‘unsigned int Antares::Solver::FinalLevelValidator::year_’ [-Wreorder]
   67 |     unsigned int year_ = 0;
      |                  ^~~~~
/home/florian/Antares_Simulator/src/solver/hydro/management/finalLevelValidator.cpp:33:1: warning:   when initialized here [-Wreorder]
   33 | FinalLevelValidator::FinalLevelValidator(
      | ^~~~~~~~~~~~~~~~~~~
[238/531] Building CXX object solver/lps/CMakeFiles/lps.dir/LpsFromAntares.cpp.o
/home/florian/Antares_Simulator/src/solver/lps/LpsFromAntares.cpp: In member function ‘const Antares::Solver::WeeklyDataFromAntares& Antares::Solver::LpsFromAntares::weeklyData(Antares::Solver::WeeklyProblemId) const’:
/home/florian/Antares_Simulator/src/solver/lps/LpsFromAntares.cpp:46:16: warning: returning reference to temporary [-Wreturn-local-addr]
   46 |         return WeeklyDataFromAntares(); // TODO Better error handling
      |                ^~~~~~~~~~~~~~~~~~~~~~~
[348/531] Building CXX object solver/optimisation/CMakeFiles/model_antares.dir/HebdoProblemToLpsTranslator.cpp.o
/home/florian/Antares_Simulator/src/solver/optimisation/HebdoProblemToLpsTranslator.cpp: In member function ‘Antares::Solver::ConstantDataFromAntares Antares::Solver::HebdoProblemToLpsTranslator::commonProblemData(const PROBLEME_ANTARES_A_RESOUDRE*) const’:
/home/florian/Antares_Simulator/src/solver/optimisation/HebdoProblemToLpsTranslator.cpp:88:38: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘std::vector<int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   88 |     if (problem->NombreDeContraintes > problem->IndicesDebutDeLigne.size())
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/florian/Antares_Simulator/src/solver/optimisation/HebdoProblemToLpsTranslator.cpp:94:38: warning: comparison of integer expressions of different signedness: ‘const int’ and ‘std::vector<int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
   94 |     if (problem->NombreDeContraintes > problem->NombreDeTermesDesLignes.size())
      |         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/src/solver/hydro/management/finalLevelValidator.cpp
+++ b/src/solver/hydro/management/finalLevelValidator.cpp
@@ -40,14 +40,14 @@ FinalLevelValidator::FinalLevelValidator(
   const unsigned int lastSimulationDay,
   const unsigned int firstMonthOfSimulation,
   HydroErrorsCollector& errorCollector):
-    hydro_(hydro),
-    areaName_(areaName),
-    areaIndex_(areaIndex),
-    initialLevel_(initialLevel),
-    finalLevel_(finalLevel),
     year_(year),
     lastSimulationDay_(lastSimulationDay),
     firstMonthOfSimulation_(firstMonthOfSimulation),
+    hydro_(hydro),
+    areaIndex_(areaIndex),
+    areaName_(areaName),
+    initialLevel_(initialLevel),
+    finalLevel_(finalLevel),
     errorCollector_(errorCollector)
 {
 }

--- a/src/solver/optimisation/HebdoProblemToLpsTranslator.cpp
+++ b/src/solver/optimisation/HebdoProblemToLpsTranslator.cpp
@@ -85,13 +85,13 @@ ConstantDataFromAntares HebdoProblemToLpsTranslator::commonProblemData(
         throw WeeklyProblemTranslationException("ConstraintesCount must be strictly positive");
     }
 
-    if (problem->NombreDeContraintes > problem->IndicesDebutDeLigne.size())
+    if (problem->NombreDeContraintes > (int)problem->IndicesDebutDeLigne.size())
     {
         throw WeeklyProblemTranslationException(
           "ConstraintesCount exceed IndicesDebutDeLigne size");
     }
 
-    if (problem->NombreDeContraintes > problem->NombreDeTermesDesLignes.size())
+    if (problem->NombreDeContraintes > (int)problem->NombreDeTermesDesLignes.size())
     {
         throw WeeklyProblemTranslationException(
           "ConstraintesCount exceed NombreDeTermesDesLignes size");

